packages/service-library/src/servicelib/queued_tasks/app_server.py

```diff
@@ -0,0 +1,69 @@
+import asyncio
+import datetime
+import threading
+from abc import ABC, abstractmethod
+from asyncio import AbstractEventLoop
+from contextlib import suppress
+from typing import TYPE_CHECKING, Final
+
+from servicelib.queued_tasks.task_manager import TaskManager
+
+if TYPE_CHECKING:
+    with suppress(ImportError):
+        from fastapi import FastAPI
+    with suppress(ImportError):
+        from aiohttp.web import Application
+
+
+STARTUP_TIMEOUT: Final[float] = datetime.timedelta(minutes=1).total_seconds()
+
+
+class BaseAppServer(ABC):
+    def __init__(self) -> None:
+        self._shutdown_event: asyncio.Event | None = None
+
+    @property
+    def fastapi_app(self) -> "FastAPI":
+        raise NotImplementedError
+
+    @property
+    def aiohttp_app(self) -> "Application":
+        raise NotImplementedError
+
+    @property
+    def event_loop(self) -> AbstractEventLoop:
+        return self._event_loop
+
+    @event_loop.setter
+    def event_loop(self, loop: AbstractEventLoop) -> None:
+        self._event_loop = loop
+
+    @property
+    def task_manager(self) -> TaskManager:
+        return self._task_manager
+
+    @task_manager.setter
+    def task_manager(self, manager: TaskManager) -> None:
+        self._task_manager = manager
+
+    @abstractmethod
+    async def on_startup(self) -> None:
```

Comment:
this one and on_shutdown look pretty specific to the FastAPI runtime, I would avoid having it in the common interface.